Add the possibility to add messages to topic.

reb3r opened the pull request:
- Also deleted some old code
- increase the buffer size of the MulticastReceiver

#2

reb3r committed:
Add the possibility to add messages to topic.

- Also deleted some old code
- increase the buffer size of the MulticastReceiver

David-Heim commented on src/app/App.java:
```diff
@@ -74,14 +60,65 @@ public static void main(String[] args) throws Exception {
     }
+
+    private static Topic topicCliChooser() throws IOException {
+        System.out.println("To which topic you want to add an message?");
```
```suggestion
        System.out.println("To which topic do you want to add a message?");
```

David-Heim commented on src/app/App.java:
```diff
@@ -74,14 +60,65 @@ public static void main(String[] args) throws Exception {
+            Topic element = it.next();
+            int index = App.topics.indexOf(element);
+            System.out.println(index + ": " + element.getName());
+        }
```
Hier bietet sich eine foreach-Schleife an. Z.B. so:
        int index;
        for (Topic topic : App.topics) {
            index = App.topics.indexOf(topic);
            System.out.println(index + ": " + topic.getName());            
        }

reb3r replied:
Gibt es bis auf die kürzere Notation irgendwelche anderen Vorteile?

David-Heim replied:
Nein, es ist vor allem kürzer. Das Auslagern der index-Variable vor die Schleife ist bei beiden Schleifen-Typen sinnvoll.

reb3r committed:
Update src/app/App.java

Co-Authored-By: David-Heim <David-Heim@users.noreply.github.com>

reb3r committed:
Added counter and some other stuff.

removed the vector clock.

reb3r committed:
Merge branch 'issue2' of https://github.com/reb3r/DBE14_WS1920_Group4 into issue2

reb3r committed:
Implemented RetransmissionRequest and usage of it.

David-Heim committed:
Added topic subscription and more

Added topic subscription and classes for leader and ring functionality.

reb3r committed:
Added HoldBackQueue as class.

Refined delivery of request and fixed retransmission of requests.

David-Heim committed:
Ringbildung implementiert

Ringbildung pro Topic implementiert. Habe zur Sicherheit schon mal comittet, ein Test steht allerdings noch aus.

David-Heim committed:
Some small fixes

David-Heim committed:
Implemented leader election

Implemented leader election using the LCR algorithm.

reb3r committed:
Improve stopping of receiver thread and formatted some classes

reb3r committed:
Refine comments and some minor  improvements

reb3r committed:
Refined Logging

reb3r committed:
Add logging

reb3r committed:
Identify sender through uuid

David-Heim committed:
Various changes including corrections and topic unsubscribe

- added command to exit topic chooser
- added possibility to unsubscribe from topics
- made some fixes to the election algorithm
- various other fixes

David-Heim committed:
Added topic delete and modify
